### PR TITLE
#599 New date editor tests

### LINKED ISSUE #599: Date Editor Coverage
Increase the date editor coverage to 90% +

### PR DESCRIPTION
**Pull request by Lipata**
Closes #599 .  

Add new date editor tests and fix related issues.

### CODE REVIEW

File: src/js/modules/infragistics.ui.editors.js [hunk 1]
```diff
@@ -10345,15 +10359,37 @@
...
+			}
+		},
+		_isNewPeriodLimited: function(name, value, delta, date) {
+			var isLimited = false;
+			if (this.options.limitSpinToCurrentField) {
+				if (delta < 0) {
+					if (value < 0) {
+						isLimited = true;
+					}
+				} else {
```
Comment (bazal4o) [inline src/js/modules/infragistics.ui.editors.js]: This statement could be converted to else if

Comment (Lipata) [inline src/js/modules/infragistics.ui.editors.js]: Done. Thanks!

**Comment by kdinev** [comment]
@Lipata I have a question. I see that there are code changes. Do those come because of issues found while new tests were being added, or are those separate issues fixed in this PR?

**Comment by Lipata** [comment]
@kdinev Those are problems found, while adding tests and they have not been logged as bugs, but instead  fixed inside this PR.

**Comment by kdinev** [comment]
@Lipata OK. Merging!

**Pull request merged**